pkgs/native_assets_cli/lib/src/config.dart

```diff
@@ -404,6 +405,58 @@ class BuildOutputBuilder extends HookOutputBuilder {
       syntax.BuildOutput.fromJson(super._syntax.json);
 }
 
+extension AddDataAssetsDirectoryExtension on BuildOutputBuilder {
+  /// Extension on [BuildOutput] to handle data asset directories and files.
+  ///
+  /// This extension provides a convenient way for build hooks to add
+  /// [DataAsset] dependencies from one or more directories or individual files.
```

Comment:
Why are we only adding the dependencies? And not also the data assets themselves?

I was envisioning it to roughly do:

pkgs/native_assets_builder/test_data/simple_data_asset/hook/build.dart

With the name defaulting to the full file path, with forward slashes on Windows. In Flutter, assets are always only accessible through their file path, so it makes sense here to also default the name to the file paths.